File: Data Structures & Algorithms/last-stone-weight/submission-1.java
class Solution {
    public int lastStoneWeight(int[] stones) {
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        int first_largest;
        int second_largest;
        int res = 0;
        for(int i:stones){
            pq.add(-i);
        }
        while(pq.size()>1){
            first_largest = Math.abs(pq.poll());
            second_largest = Math.abs(pq.poll());
            System.out.println(first_largest+" "+second_largest);
            if(first_largest!=second_largest){
                pq.offer(-(first_largest-second_largest));
            }
        }
        if(!pq.isEmpty()){
            res = pq.peek();
        }
        return Math.abs(res);
    }
}

File: Data Structures & Algorithms/three-integer-sum/submission-1.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        nums.sort()
        i = 0
        se = set()
        res = []
        while i<len(nums)-2:
            j = i+1
            k = len(nums)-1
            while j<k:
                if nums[i]+nums[j]+nums[k]==0:
                    if (nums[i],nums[j],nums[k]) not in se:
                        res.append([nums[i],nums[j],nums[k]])
                        se.add((nums[i],nums[j],nums[k]))
                    j+=1
                elif nums[i]+nums[j]+nums[k]>0:
                    k-=1
                else:
                    j+=1
            i+=1
        return res
